### PR TITLE
Fot DHCP reflection changes

### PR DESCRIPTION
**Pull request by anurag6**

### CODE REVIEW

File: testing/python_lib/integration_base.py [hunk 1]
```diff
@@ -118,8 +126,12 @@ def _write_faucet_config(self, config):
         return self._write_yaml_to_file(filename, config)
 
     def _get_faucet_config_path(self):
-        return os.path.dirname(os.path.abspath(__file__)) + \
-            '/../../inst/forch-faucet-1/faucet/faucet.yaml'
+        if self.stack_options.get('fot'):
+            return os.path.dirname(os.path.abspath(__file__)) + \
+                '/../../inst/forch-controller-1/faucet/faucet.yaml'
```
Comment (grafnu) [inline testing/python_lib/integration_base.py:131]: replace common string with format placeholder constant (../inst/%s/) and then use % substitution

File: testing/python_lib/test_fot.py [hunk 2]
```diff
@@ -215,5 +216,40 @@ def test_ports_states(self):
...
 
+class FotContainerTest(IntegrationTestBase):
+    """Test suite for dynamic config changes"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.stack_options['fot'] = True
+        self.stack_options['local'] = True
+
+    def tearDown(self):
```
Comment (grafnu) [inline testing/python_lib/test_fot.py]: remove function entirely

File: etc/FAUCET_VERSION [hunk 3]
```diff
@@ -1 +1 @@
-1.9.48
+master
```
Comment (grafnu) [inline etc/FAUCET_VERSION:1]: I think the requisite change is in 1.9.50 that's currently available, so if that works we should just update here and make sure it all still works. I'm in the middle of testing 1.9.50 integration and it all seems good, but might as well get that tested with this too.

File: etc/start_faux [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 echo Running faux $*
 
+umount /etc/resolv.conf
```
Comment (grafnu) [inline etc/start_faux:5]: add comment as to WHY this is here.

File: testing/python_lib/test_fot.py [hunk 5]
```diff
@@ -215,5 +216,40 @@ def test_ports_states(self):
...
+        pass
+
+    def test_dhcp_reflection(self):
+        """Test to check DHCP reflection when on test VLAN"""
+        def dhclient_method(container=None):
```
Comment (grafnu) [inline testing/python_lib/test_fot.py:229]: Either use positional args or keyword, but try not to mix them. E.g., here you're using a keyword arg (container=), but when you call it you're just using positional i.e. dhclient_metho('forch-faux-1')  -- just need to fix one of the two places to match the other.

**Review by grafnu — COMMENTED** [review]
Nice... getting close!

File: testing/python_lib/integration_base.py (hunk 1 above)
Comment (anurag6) [reply testing/python_lib/integration_base.py]: Addressed.

File: testing/python_lib/test_fot.py (hunk 2 above)
Comment (anurag6) [reply testing/python_lib/test_fot.py]: Done.

File: etc/FAUCET_VERSION (hunk 3 above)
Comment (anurag6) [reply etc/FAUCET_VERSION]: changed to 1.9.50

File: etc/start_faux (hunk 4 above)
Comment (anurag6) [reply etc/start_faux]: Done

File: testing/python_lib/test_fot.py (hunk 5 above)
Comment (anurag6) [reply testing/python_lib/test_fot.py:229]: Done

File: testing/python_lib/integration_base.py [hunk 6]
```diff
@@ -128,10 +130,9 @@ def _write_faucet_config(self, config):
     def _get_faucet_config_path(self):
         if self.stack_options.get('fot'):
             return os.path.dirname(os.path.abspath(__file__)) + \
-                '/../../inst/forch-controller-1/faucet/faucet.yaml'
-        else:
-            return os.path.dirname(os.path.abspath(__file__)) + \
-                '/../../inst/forch-faucet-1/faucet/faucet.yaml'
+                ('/../../inst/%s/faucet/faucet.yaml' % ('forch-controller-1'))
```
Comment (grafnu) [inline testing/python_lib/integration_base.py:133]: extract shared string to constant

File: etc/start_faux [hunk 7]
```diff
@@ -2,7 +2,7 @@
 
 echo Running faux $*
 
-umount /etc/resolv.conf
+umount /etc/resolv.conf # Necessary to run dhclient in docker containers
```
Comment (grafnu) [inline etc/start_faux:5]: comment goes on previous line not trailing (just style convention)

Comment (anurag6) [reply etc/start_faux]: Resolved.

File: testing/python_lib/integration_base.py (hunk 6 above)
Comment (anurag6) [reply testing/python_lib/integration_base.py]: changed.

File: etc/start_faux [hunk 8]
```diff
@@ -2,8 +2,8 @@
 
 echo Running faux $*
 
-umount /etc/resolv.conf # Necessary to run dhclient in docker containers
-
+# Necessary to run dhclient in docker containers
+umount /etc/resolv.conf
```
Comment (grafnu) [inline etc/start_faux:6]: now you lost the blank space after - it's a separate logical construct from the things that follow it

File: testing/python_lib/integration_base.py [hunk 9]
```diff
@@ -128,11 +128,12 @@ def _write_faucet_config(self, config):
         return self._write_yaml_to_file(filename, config)
 
     def _get_faucet_config_path(self):
+        config_file = '/../../inst/%s/faucet/faucet.yaml'
```
Comment (grafnu) [inline testing/python_lib/integration_base.py:131]: config_file_format

File: etc/start_faux (hunk 8 above)
Comment (anurag6) [reply etc/start_faux:6]: changed

File: testing/python_lib/integration_base.py (hunk 9 above)
Comment (anurag6) [reply testing/python_lib/integration_base.py]: changed.

**Review by grafnu — APPROVED** [review]

**Pull request merged**